### ISSUE TITLE
initial app load delay

### ISSUE THREAD

**Issue by murali-shris** [atsign-foundation/at_client_sdk#288]
Initial load in app is taking more than 10 seconds. Probable issue is waiting for monitor response

**Comment by yahu1031** [comment]
@kalluriramkumar As I have debugged it and I found `_getQueueResponse()` takes the time delay.
```dart
///Returns the response of the monitor verb queue.
  Future<String> _getQueueResponse() async {
    var maxWaitMilliSeconds = 5000;
    var monitorResponse;
    //wait maxWaitMilliSeconds seconds for response from remote socket
    var loopCount = (maxWaitMilliSeconds / 50).round();
    for (var i = 0; i < loopCount; i++) {
      await Future.delayed(Duration(milliseconds: 90));
      var queueLength = _monitorVerbResponseQueue.length;
      if (queueLength > 0) {
        // result from another secondary is either data or a @<atSign>@ denoting complete
        // of the handshake
        monitorResponse = _defaultResponseParser.parse(_monitorVerbResponseQueue.removeFirst());
      }
    }
    return monitorResponse.response;
  }
  ```
In this function
```dart
var loopCount = (maxWaitMilliSeconds / 50).round();
``` 
will have a value of 100 and the `for` loop will iterate almost 100 times. So the delay in the for loop will become 90 * 100 ms = 9sec. That is where the delay happens and this function is being called twice so we are getting 20sec delay including the other operations. Here are few logs 

```log
I/flutter ( 3499): FINER|2021-09-20 13:17:39.046337|AtLookup|AtLookup.findSecondary received atsign: @gothic681
I/flutter ( 3499): FINER|2021-09-20 13:17:40.147201|AtLookup|AtLookup.findSecondary got answer: bfef58ac-703d-536d-aa43-d0904937db03.buzz.do-sf2.atsign.zone:3093 and closing connection
I/flutter ( 3499): FINER|2021-09-20 13:17:40.482303|AtLookup|AtLookup.findSecondary received atsign: @gothic681
I/flutter ( 3499): FINER|2021-09-20 13:17:41.487231|AtLookup|AtLookup.findSecondary got answer: bfef58ac-703d-536d-aa43-d0904937db03.buzz.do-sf2.atsign.zone:3093 and closing connection
I/flutter ( 3499): FINER|2021-09-20 13:17:51.752829|Monitor|Authenticating the monitor connection: from result:_ec8c2bd4-2221-44a7-b512-e7459ba48309@gothic681:1477f217-4443-41d4-9e20-3dbd6768dff6
I/flutter ( 3499): FINER|2021-09-20 13:17:51.935610|Monitor|Authenticating the monitor connection: pkam:TOAj41B84H2U2B3fyvlu2/q/O1/ogPnA+xrmBBDd7ZqwUTwgquKZlb6A/hEfFZwwbNwfE2KDPjNPvMSVjwyazeuRbtUO/wDpZn2s6qgpyWk9VOkvn1TwW3H0p0IalIiAtWocIPq6IA1zU0H5SCnR/EF2wwToAbKmoMEVnMroaU1SeP00yypQoFFoN09bGY/I0tOt1AGLHb4Zv226fMXWetRWwgbItoX5tILqIpfnTgcbEUtPt0dw7xm7hQc/TaLTPzem+1tQwOYDfUbWd0kbXwctiRJqRZsn1siejKUhRN299nkMhiRH22AyDtIywmaNvTtvjcwYnvX7Kz99iJXVQA==
I/flutter ( 3499): FINER|2021-09-20 13:18:01.454750|Monitor|Monitor connection authentication successful
```

Time delays are here `13:17:41` - `13:17:51` - `13:18:01`

**Comment by murali-shris** [comment]
Issue fixed with at_client v3.0.1
